### PR TITLE
webutil: replace cloneValue impl with serialize/deserialize

### PR DESCRIPTION
**Pull request by EladKeyshawn**
From #10208

### CODE REVIEW

File: runtime/js/01_web_util.js [hunk 1]
```diff
@@ -35,87 +35,7 @@
...
-        }
-        Reflect.setPrototypeOf(clonedObj, Reflect.getPrototypeOf(value));
-        return clonedObj;
-      }
-      case "symbol":
-      case "function":
-      default:
-        throw new DOMException("Uncloneable value in stream", "DataCloneError");
-    }
+    return Deno.core.deserialize(Deno.core.serialize(value));
```
Comment (evanwashere) [inline runtime/js/01_web_util.js:38]: should retain type checks and manual cloning for primitive values and ArrayBuffer/TypedArray

**Review by CrimsonCodes0 — COMMENTED** [review]
The implementation should still throw the old DOMException, serialize/deserialize throw something else.

File: runtime/js/01_web_util.js (hunk 1 above)
Comment (lucacasonato) [reply runtime/js/01_web_util.js:38]: As a performance optimisation?

Comment (evanwashere) [reply runtime/js/01_web_util.js:38]: yeah, Deno.core.serialize allocates a new buffer, so it will be big overhead for primitives and ArrayBuffer

Comment (bartlomieju) [reply runtime/js/01_web_util.js]: Reverted back

**Review by lucacasonato — APPROVED** [review]

**Pull request merged**